libcloud/compute/drivers/scaleway.py

```diff
@@ -0,0 +1,615 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Scaleway Driver
+"""
+
+import copy
+try:
+    import simplejson as json
+except:
+    import json
+
+from libcloud.common.base import ConnectionUserAndKey, JsonResponse
+from libcloud.compute.base import NodeDriver, NodeImage, Node, NodeSize
+from libcloud.compute.base import NodeLocation
+from libcloud.compute.base import StorageVolume, VolumeSnapshot, KeyPair
+from libcloud.compute.providers import Provider
+from libcloud.compute.types import NodeState, VolumeSnapshotState
+from libcloud.utils.iso8601 import parse_date
+from libcloud.utils.py3 import httplib
+
+__all__ = [
+    'ScalewayResponse',
+    'ScalewayConnection',
+    'ScalewayNodeDriver'
+]
+
+SCALEWAY_API_HOSTS = {
+    'default': 'api.scaleway.com',
+    'account': 'account.scaleway.com',
+    'par1': 'cp-par1.scaleway.com',
+    'ams1': 'cp-ams1.scaleway.com',
+}
+
+# The API doesn't currently expose all of the required values for libcloud,
+# so we simply list what's available right now, along with all of the various
+# attributes that are needed by libcloud.
```

Comment:
How did you retrieve these data? The ideal scenario would be to have a script in the `contrib` directory to scrape these prices. We would then run this script before every release.

Reply:
This was filled manually, unfortunately. It might be scrapable, except for disk sizes, which it doesn't report anywhere I've seen. I'm going to be contacting them about exposing the hard-coded info in their API instead, so we don't even need to have a scraper hack in place.

Reply:
Okay, thank you! This should not block this PR, however.

Reply:
Fixed in b6356bb. :slightly_smiling_face: